### ISSUE TITLE
Misleading error message when an invalid y column name is provided to px.line

### ISSUE THREAD

**Issue by joshHug** [plotly/plotly.py#4040]
When attempting to plot multiple y values using px.line (or similar), erroneous column names lead to a misleading error message that states that the length of the arguments is incorrect, when in fact the problem is that there the column name is wrong.

Simple to reproduce:

```
import plotly.express as px
import pandas as pd
import numpy as np
x = np.linspace(0, 10, 100)
data1 = x**2
data2 = x**3
df = pd.DataFrame({"x": x, "data1": data1, "data2": data2})
px.line(df, x = "x", y = ["data1", "data2"]) # works fine
px.line(df, x = "x", y = ["data1", "Data2"]) 

# this last line gives a misleading error message: 
# ValueError: All arguments should have the same length. The length of argument `y` is 2, 
# whereas the length of  previously-processed arguments ['x'] is 100
```

**Comment by nicolaskruchten** [comment]
Yes, we should definitely find a way to give you a better hint there that there's a typo in the contents of `y`. It's a bit tricky to do and we're love some help to figure out how to do it! The relevant code is near here: https://github.com/plotly/plotly.py/blob/master/packages/python/plotly/plotly/express/_core.py#L1247

**Comment by nicolaskruchten** [comment]
The reason it's tricky is that we land deep in that `if` block based on some determination that happened way ahead of time here https://github.com/plotly/plotly.py/blob/master/packages/python/plotly/plotly/express/_core.py#L1315 ... basically up front PX decides if it's in "wide mode" based on whether or not the contents of `y` are all column names, and then much later if they're not, it checks the length of the data `y` points to against the length of the data `x` points to.

**Comment by nicolaskruchten** [comment]
See also https://github.com/plotly/plotly.py/issues/3359 and https://github.com/plotly/plotly.py/issues/3474 and https://github.com/plotly/plotly.py/issues/2586

**Comment by abarpan3** [comment]
Hi, was just exploring this issue. Seem to have found something interesting. 
The function _is_col_list()_ is built to check if the list passed to it is a list of columns.

This function is initally used to check if dataframe is wide-mode or not inside the build dataframe function.
https://github.com/plotly/plotly.py/blob/master/packages/python/plotly/plotly/express/_core.py#L1340

 
In this function, there is a check to see if the column actually is in the Dataframe _(Line 1036)_. And this issue can be highlighted out easily at that point, if a column name is not present giving the user a more helpful error message.

https://github.com/plotly/plotly.py/blob/master/packages/python/plotly/plotly/express/_core.py#L1036

A potential change could be:
![image](https://github.com/plotly/plotly.py/assets/26029272/c253dfec-b91d-4ba5-a0ed-cf44d1e6380e)

**Comment by abarpan3** [comment]
Hi, was exploring this issue a bit. So the _build_dataframe_ function has another function _is_col_list_ to check if the supplied argument for x/y is a list or not (hence wide-form) https://github.com/plotly/plotly.py/blob/master/packages/python/plotly/plotly/express/_core.py#L1339

Now this _is_col_list_ function has a check inside it to see if a column is present in the dataframe, and if not it returns _wide_mode_ as false. 
https://github.com/plotly/plotly.py/blob/master/packages/python/plotly/plotly/express/_core.py#L1036

Here, in the _is_col_list_ functionwe can catch the erroneous column name and give the user a more helpful error message. So I tried to make 2 changes,

1. **Line 1035** : Checking if y value is of pd.Series type and returning _wide_mode_ as false, if the condition is fulfilled. This takes care of any direct array inputs in x/y arguments like below example.
`px.scatter(df1,x = df1.sales, y = df2.market)`

2. **Line 1040 - ** : Checking additionally if col_name supplied in list is present in df. And returning a more helpful error message. 

![image](https://github.com/plotly/plotly.py/assets/26029272/1e4f898a-f2dd-4fe4-8fcb-1b97f76d5cc8)


Could anyone kindly provide any feedback if this seems like the right approach?
